src/panels/lovelace/cards/energy/hui-energy-gas-graph-card.ts

```diff
@@ -270,6 +274,12 @@ export class HuiEnergyGasGraphCard
         (source) => source.type === "gas"
       ) as GasSourceTypeEnergyPreference[];
 
+    const gasStatisticIdsWithMeta: StatisticsMetaData[] =
```

Comment:
We already fetch the meta when we generate the energy data: <https://github.com/home-assistant/frontend/blob/dev/src/data/energy.ts#L292> let's use that instead of fetching it again.

Reply:
We know what stat ids we need, as that are the ones we fetch the statistics for, so I don't think that is an issue?

Reply:
somethink like this in energy.ts?:
```js
for (const source of prefs.energy_sources) {
    if (source.type === "solar" || source.type === "gas") {
      if (source.stat_energy_from){
        statIDMetasToFetch.push(source.stat_energy_from)
      }
    } else if (source.type === 'grid'){
      for (const flowFrom of source.flow_from) {
        statIDMetasToFetch.push(flowFrom.stat_energy_from)
      }
    }
  }
  const allStatisticIdsWithMeta: StatisticsMetaData[] =
    await getStatisticMetadata(
      hass,
      statIDMetasToFetch
    );
```

Reply:
I would just fetch the metadata for all statistics we fetch:

```ts
const metadata =
    await getStatisticMetadata(
      hass,
      statIDs
    );
```

and add that to `EnergyData`.

Reply:
I think it's done. Let me know if it's ok. I have only data for gas entities in database, so I can't test all changes I've done :/